test/Microsoft.ML.Core.Tests/UnitTests/TestResourceDownload.cs

```diff
@@ -0,0 +1,276 @@
+            Environment.SetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable, baseDirUri.AbsoluteUri);
+            var envVar = Environment.GetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable);
+            Assert.True(envVar == baseDirUri.AbsoluteUri);
+            var path = DeleteOutputPath(Path.Combine("resources", "subdir"), "breast-cancer.txt");
+
+            var bc = GetDataPath("breast-cancer.txt");
+            File.Copy(bc, path);
+
+            var saveToDir = GetOutputPath("copyto");
+            DeleteOutputPath("copyto", "breast-cancer.txt");
+            var sbOut = new StringBuilder();
+            var env = new ConsoleEnvironment(42);
+            using (var ch = env.Start("Downloading"))
+            {
+                try
+                {
+                    var t = ResourceManagerUtils.Instance.EnsureResourceAsync(env, ch, "subdir/breast-cancer.txt", "breast-cancer.txt", saveToDir, 1 * 60 * 1000);
+                    t.Wait();
+
+                    if (t.Result.ErrorMessage != null)
+                        Fail(String.Format("Expected zero length error string. Received error: {0}", t.Result.ErrorMessage));
+                    if (t.Status != TaskStatus.RanToCompletion)
+                        Fail("Download did not complete succesfully");
+                    if (!File.Exists(GetOutputPath("copyto", "breast-cancer.txt")))
+                    {
+                        Fail($"File '{GetOutputPath("copyto", "breast-cancer.txt")}' does not exist. " +
+                            $"File was downloaded to '{t.Result.FileName}' instead." +
+                            $"MICROSOFTML_RESOURCE_PATH is set to {Environment.GetEnvironmentVariable(Utils.CustomSearchDirEnvVariable)}");
+                    }
+                    Done();
+                }
+                finally
+                {
+                    // Set environment variable back to its old value.
+                    Environment.SetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable, envVarOld);
+                    Environment.SetEnvironmentVariable(Utils.CustomSearchDirEnvVariable, resourcePathVarOld);
+                }
+            }
+        }
+
+        [Fact]
+        [TestCategory("ResourceDownload")]
+        public void TestDownloadError()
+        {
+            var envVarOld = Environment.GetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable);
+            var timeoutVarOld = Environment.GetEnvironmentVariable(ResourceManagerUtils.TimeoutEnvVariable);
+            var resourcePathVarOld = Environment.GetEnvironmentVariable(Utils.CustomSearchDirEnvVariable);
+            Environment.SetEnvironmentVariable(Utils.CustomSearchDirEnvVariable, null);
+
+            // Bad path.
+            try
+            {
+                if (!Uri.TryCreate($@"\\ct01\public\{Guid.NewGuid()}\", UriKind.Absolute, out var badUri))
```

Comment:
I'm not sure how the CI nodes will handle a Windows share that it, hopefully, can't access. Your internal machine may react differently.

Reply:
This works on both local and CI nodes in my experience, but I have changed it so that there are no Windows share directories involved.

Reply:
Did you push the commit w/ the change? I still see `$@"\\ct01\public\{Guid.NewGuid()}\"`.